packages/realm/src/Object.ts

```diff
@@ -93,15 +96,11 @@ class RealmObject<T = DefaultObject> {
         const defaultValue = helpers.properties.get(property.name).default;
         if (typeof defaultValue !== "undefined") {
           result[property.name] = defaultValue;
-        } else if (
-          !(property.type & binding.PropertyType.Collection) &&
-          !(property.type & binding.PropertyType.Nullable) &&
-          created
-        ) {
-          throw new Error(`Missing value for property '${property.name}'`);
-        }
+        } else if (created) {
+        throw new Error(`Missing value for property '${property.name}'`);
```

Comment:
Do you remember why this was needed?

Reply:
this was a mess-up on my end from rebasing. I'll revert this.